src/core/stack.js

```diff
@@ -444,7 +456,9 @@ export default class Stack {
      * @instance  
      */
     Query() {
-        let query = new Query();
+        let query = (this.type === "taxonomy" || this.type === "contentType") ?
+        new TaxonomyQuery() :
```

Comment:
@ishaileshmishra I've added this because content-type API also supports taxonomy queries. I've added a comment in the code, explaining the reason for this.